pyracf/resource/resource_admin.py

```diff
@@ -216,6 +270,215 @@ def get_my_access(self, resource: str, class_name: str) -> Union[str, bytes, Non
+        """Extract the attributes of a custom field."""
+        full_profile_name = f"{custom_field_type}.csdata.{custom_field_name}"
+        profile = self.extract(
+            resource=full_profile_name,
+            class_name="CFIELD",
+            segments=["cfdef"],
+            profile_only=True,
+        )
+        return profile["cfdef"]
+
+    def delete_custom_field(
+        self, custom_field_name: str, custom_field_type: str
+    ) -> Union[dict, bytes]:
+        """Delete a custom field."""
+        full_profile_name = f"{custom_field_type}.csdata.{custom_field_name}"
+        return self.delete(resource=full_profile_name, class_name="CFIELD")
+
+    # ============================================================================
+    # Kerberos Realm Administration
+    # ============================================================================
+    def add_kerberos_realm(
+        self, kerberos_realm_name: str, traits: dict = {}
+    ) -> Union[dict, bytes]:
+        """Create a new kerberos realm profile."""
+        return self.add(resource=kerberos_realm_name, class_name="REALM", traits=traits)
+
+    def alter_kerberos_realm(
+        self, kerberos_realm_name: str, traits: dict = {}
+    ) -> Union[dict, bytes]:
+        """Alter an existing kerberos realm profile."""
+        return self.alter(
+            resource=kerberos_realm_name, class_name="REALM", traits=traits
+        )
+
+    def extract_kerberos_realm(self, kerberos_realm_name: str) -> Union[dict, bytes]:
+        """Extract the attributes of a kerberos realm profile."""
+        profile = self.extract(
+            resource=kerberos_realm_name,
+            class_name="REALM",
+            segments=["kerb"],
+            profile_only=True,
+        )
+        return profile["kerb"]
+
+    def delete_kerberos_realm(self, kerberos_realm_name: str) -> Union[dict, bytes]:
+        """Delete a kerberos realm profile."""
+        return self.delete(resource=kerberos_realm_name, class_name="REALM")
+
+    # ============================================================================
+    # Signed Program Administration
+    # ============================================================================
+    def add_signed_program(
+        self, signed_program_name: str, traits: dict = {}
+    ) -> Union[dict, bytes]:
+        """Create a new signed program profile."""
+        if "sigver:library" in traits:
+            traits["base:member"] = traits["sigver:library"]
+            del traits["sigver:library"]
+        return self.add(
+            resource=signed_program_name, class_name="PROGRAM", traits=traits
+        )
+
+    def alter_signed_program(
+        self, signed_program_name: str, traits: dict = {}
+    ) -> Union[dict, bytes]:
+        """Alter an existing signed program profile."""
+        if "sigver:library" in traits:
+            traits["base:member"] = traits["sigver:library"]
+            del traits["sigver:library"]
```

Comment:
What is this for?

Doesn't appear to be covered in the code coverage report.

Reply:
That's correct, I didn't write full unit tests for signed program administration as its experimental, but I wanted to build out the full code fully. I designed these subfunctions after their xml interface counterparts even though we are not using those interfaces directly.

If you review the interface for signed program (https://www.ibm.com/docs/en/zos/3.1.0?topic=types-signed-program). _The sigrequired, failload, and sigaudit fields correspond to the sigreqd, failload, sigaudit fields in the SIGVER segment of general resource profiles, as described in the [General resource administration](https://www.ibm.com/docs/en/SSLTBW_3.1.0/com.ibm.zos.v3r1.ichd100/resprof.htm) in [Reference documentation tables](https://www.ibm.com/docs/en/SSLTBW_3.1.0/com.ibm.zos.v3r1.ichd100/tabref.htm). The library field corresponds to the member field in the BASE segment._ I can remove this for now, but I thought that it would be better to build in the functionality now and debug/test later as we move to make the feature non-experimental.

Reply:
Thank you for the explanation. I just wanted to understand what this is for. Just make sure that the documentation correctly makes the distinction between stable and experimental features and we should be good.